packages/flutter/lib/src/material/debug.dart

```diff
@@ -66,3 +67,63 @@ bool debugCheckHasMaterial(BuildContext context) {
   }());
   return true;
 }
+
+
+/// Asserts that the given context has a [Localizations] ancestor that contains
+/// a [MaterialLocalizations] delegate.
+///
+/// Used by many material design widgets to make sure that they are
+/// only used in contexts where they have access to localizations.
+///
+/// To call this function, use the following pattern, typically in the
+/// relevant Widget's build method:
+///
+/// ```dart
+/// assert(debugCheckHasMaterialLocalizations(context));
+/// ```
+///
+/// Does nothing if asserts are disabled. Always returns true.
+bool debugCheckHasMaterialLocalizations(BuildContext context) {
+  assert(() {
+    if (Localizations.of<MaterialLocalizations>(context, MaterialLocalizations) == null) {
+      final StringBuffer message = new StringBuffer();
+      message.writeln('No MaterialLocalizations found.');
+      message.writeln(
+          '${context.widget.runtimeType} widgets require MaterialLocalizations '
+              'to be provided by a Localizations widget ancestor.'
+      );
```

Comment:
indentation is inconsistent here. generally we prefer two spaces indent throughout (as in other parts of the file)